lib/language_pack/cache.rb

```diff
@@ -42,6 +42,41 @@ def load(path, dest = nil)
     copy (@cache_base + path), dest
   end
 
+  # Store a path as an archive, which is more efficient for directories
+  # with a lot of files (e.g. asset caches).
+  # @param [String] path relative directory to store as an archive
+  def store_archive(path)
+    return false unless File.exist?(path)
+    tar = archive_path(path)
+    tar.delete if tar.exist?
+    FileUtils.mkdir_p File.dirname(tar)
+    system("tar -cf #{tar} #{path}")
+  end
+
+  # Store a path as an archive if it has changed, based on directory size as
+  # reported by `du`, which is not a guarantee. Use only where necessary with
+  # large cache directories that won't cause problems if skipped.
+  # @param [String] path relative directory to store as an archive
+  # @param [Integer] previous_size the previous size of the directory from `load_archive`
+  def store_archive_if_changed(path, previous_size)
```

Comment:
Would it be possible to use an MD5 hash instead? It seems like size could lead to some false negatives.

Reply:
I tried hashing the directory contents of `node_modules`, but it was slower than the time it takes to transfer them.

Reply:
lol yeah good point.

Reply:
100% open to a smarter way of doing this, though.. maybe hashing mtimes?

Reply:
Committed 7359f54c7c09ac7f249855cb9554c851c5d71737, let me know if you think that mtime+filename hashing is an improvement. Performance-wise, it's about 4x slower than `du -sb` - `node_modules` takes 4-5s.

Very little performance difference with/without `awk`, or with `md5` instead. Our `node_modules` is currently 139,912 files, so it's literally just `ls` that takes so long.

Reply:
I'm pretty sure they're just attaching an EBS to an ephemeral EC2 instance, so the EC2 instance gets torn down after the build is finished but they just reattach the EBS to the next build instance. I imagine `tar` is about the best we can do.